src/common/utils/time/time.ts

```diff
@@ -59,44 +49,44 @@ export function exclusiveToInclusiveEnd(exclusiveEnd: Date): Date {
   return new Date(exclusiveEnd.valueOf() - 1);
 }
 
-export function formatTimeRange(timeRange: TimeRange, timezone: Timezone, displayYear: DisplayYear): string {
-  const { start, end } = timeRange;
-  const startWallTime = moment.tz(start, timezone.toString());
-  const endWallTime = moment.tz(end, timezone.toString());
-  const endWallTimeInclusive = getEndWallTimeInclusive(end, timezone);
-
-  let showingYear = true;
-  let formatted: string;
-  if (startWallTime.year() !== endWallTimeInclusive.year()) {
-    formatted = [startWallTime.format(FORMAT_WITH_YEAR), endWallTimeInclusive.format(FORMAT_WITH_YEAR)].join(" - ");
-  } else {
-    showingYear = displayYear === DisplayYear.ALWAYS ||
-      (displayYear === DisplayYear.IF_DIFF && !isCurrentYear(endWallTimeInclusive.year(), timezone));
-    const fmt = showingYear ? FORMAT_WITH_YEAR : FORMAT_WITHOUT_YEAR;
-    if (startWallTime.month() !== endWallTimeInclusive.month() || startWallTime.date() !== endWallTimeInclusive.date()) {
-      formatted = [startWallTime.format(FORMAT_WITHOUT_YEAR), endWallTimeInclusive.format(fmt)].join(" - ");
-    } else {
-      formatted = startWallTime.format(fmt);
-    }
+function formatDate(start: Moment, end: Moment, showingYear: boolean): string {
```

Comment:
I would create two separate functions instead of showingYear parameter. Even if the parameter is set to false there are situations where year is presented, weird.

Reply:
You mean `formatDateOfTimeRangeWithYear` and `formatDateOfTimeRangeWithoutYear`?